apps/src/music/player/MusicLibrary.ts

```diff
@@ -188,6 +188,13 @@ export default class MusicLibrary {
     return `${baseUrl}${this.libraryJson.path}/${folder.path}/${optionalSoundPath}${soundData.src}.mp3`;
   }
 
+  getSoundIdFromUrl(soundUrl: string): string {
+    const soundUrlParts = soundUrl.split('/');
+    const source = soundUrlParts[soundUrlParts.length - 1].slice(0, -4);
+    const folder = soundUrlParts[soundUrlParts.length - 2];
```

Comment:
Dangit. I wanted to suggest using `Array.prototype.at` here, since it can take negative offsets, but that was only added to Safari in 15.4, and we're currently supporting down to 13 (14 this summer). Chrome has had it since 92 and Firefox since 90.

It _would have been_ simplified to:

```
const source = soundUrlParts.at(-1).slice(0, -4);
const folder = soundUrlParts.at(-2);
```

So just noting it as something that could be tweaked when our min reqs catch up.

Reply:
Good to know - thanks!